### DISCUSSION TITLE
Could `useField` take a `defaultValue`?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [edmundhung/conform#436] [Ideas]
We have a lot of fieldsets that get reused between "new" forms and "edit" forms:

- In "new" forms, fields might need some initial value, like "this box should be checked by default".
- In "edit" forms, the default value instead comes from the server.

What I'd love is some way to set these initial values at the **fieldset** level, to be used only if the **form** has no `defaultValue`.

My idea was a `defaultValue` option on the `useField` hook:

- If `useForm` was called with `defaultValue === undefined`, then `useField` would use its `defaultValue`.
- If `useForm` was called with `defaultValue !== undefined`, then `useField` would ignore its `defaultValue`.

I wouldn't want or expect anything more complex than this—definitely no merging of values. This would be super simple; the `defaultValue` passed to `useField` would apply **if and only if** no `defaultValue` was passed to `useForm`.

I don't know if this suggested API is the best one, but I think this very common use case would be great (and hopefully straightforward) to support. What we're doing right now is adding a `defaultValue` property to each fieldset, then manually adding it to the `defaultValue` passed to `useForm`; the problem with this, of course, is that it's easy to forget.

Thanks!

**Comment by austinzentzkippdc** [comment]
The way I've seen this done in this and other form-helper libraries is with a function you'd pass to defaultValue.

```tsx
const [form, fields] = useForm({
  defaultValue: (user?: UserSchema) => ({
    id: user?.id ?? '',
    address: {
      street: user?.address.street ?? '',
      isPublic: user?.address.street ?? true
    }
  })
});
```

I haven't tested it, but I think that'll mean that your useField hooks will inherit whatever value you set up in the useForm hook. Does that solve your issue?

**Reply by aaronadamsCA** [reply]
I don't think so, because it still requires us to "hoist" default values up to the form component.

In our case we export a `defaultValue` from the fieldset component (if there is one), and then the form component passes it to `useForm`. But if you forget to do this, a type check won't save you. It's an implementation detail of the fieldset I'd prefer to be able to hide from the form.
